libraries/splunk_app.rb

```diff
@@ -27,6 +27,22 @@ def self.merge_hashes(*hashes)
       end
     end
   end
+
+  # Utility Class to parse app version strings
+  class AppVersion
+    def initialize(version)
+      @version = version.chomp.empty? ? nil : version.chomp unless version.nil?
+      @base, @prerelease = @version.split(' ', 2) unless @version.nil?
+      @type = @prerelease.nil? ? :base : :prerelease unless @version.nil?
```

Comment:
I could wrap the last two lines in an `unless @version.nil?` The first line is out because if `version` is only whitespace or the empty string `@version` is `nil`. 

But when I then do so, I wind up with a rubocop warning of `Use a guard clause instead of wrapping the code inside a conditional expression`